llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2198,6 +2198,10 @@ static void transformRecipestoEVLRecipes(VPlan &Plan, VPValue &EVL) {
   VPRegionBlock *LoopRegion = Plan.getVectorLoopRegion();
   VPBasicBlock *Header = LoopRegion->getEntryBasicBlock();
 
+  assert(all_of(Plan.getVF().users(),
+                IsaPred<VPVectorEndPointerRecipe, VPScalarIVStepsRecipe>));
+  Plan.getVF().replaceAllUsesWith(&EVL);
+
```

Comment:
We need to move this above the FOR transform because it introduces a trunc that uses the original VF